### PR TITLE
Added app insight alert configuration

### PR DESCRIPTION
**Pull request by Jamesnelson117**
**Before creating a pull request make sure that:**

- [ ] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -54,6 +54,41 @@ module "fpl-scheduler-db" {
...
 
+data "azurerm_key_vault_secret" "fpla_support_email_secret" {
+  name      = "fpla-support-email"
+  vault_uri = "${module.key-vault.key_vault_uri}"
+}
+
+module "fpla-action-group" {
+  source                 = "git@github.com:hmcts/cnp-module-action-group"
+  location               = "global"
+  env                    = "aat"
```
Comment (tomaszpowroznik) [inline infrastructure/main.tf:65]: ${var.env}

File: infrastructure/main.tf [hunk 2]
```diff
@@ -54,6 +54,41 @@ module "fpl-scheduler-db" {
...
+  resourcegroup_name     = "fpl-case-service-aat"
```
Comment (tomaszpowroznik) [inline infrastructure/main.tf:66]: fpl-case-service-${var.env}

File: infrastructure/main.tf [hunk 3]
```diff
@@ -54,6 +54,41 @@ module "fpl-scheduler-db" {
...
+  action_group_name      = "fpla-support"
+  short_name             = "fpla-support"
+  email_receiver_name    = "FPLA Support Mailing List"
+  email_receiver_address = "${data.azurerm_key_vault_secret.fpla_support_email_secret.value"
```
Comment (tomaszpowroznik) [inline infrastructure/main.tf:70]: no closing parenthesis

File: infrastructure/main.tf [hunk 4]
```diff
@@ -54,6 +54,41 @@ module "fpl-scheduler-db" {
...
+}
+
+module "fpla-performance-alert" {
+  source                     = "git@github.com:hmcts/cnp-module-metric-alert"
+  location                   = "${var.appinsights_location}"
+
+  app_insights_name          = "fpl-case-service-appinsights-aat"
```
Comment (tomaszpowroznik) [inline infrastructure/main.tf:77]: dynamic env

File: infrastructure/main.tf [hunk 5]
```diff
@@ -54,6 +54,41 @@ module "fpl-scheduler-db" {
...
+  short_name             = "fpla-support"
+  email_receiver_name    = "FPLA Support Mailing List"
+  email_receiver_address = "${data.azurerm_key_vault_secret.fpla_support_email_secret.value}"
+}
+
+module "fpla-performance-alert" {
+  source                     = "git@github.com:hmcts/cnp-module-metric-alert"
+  location                   = "${var.appinsights_location}"
+
+  app_insights_name          = "fpl-case-service-appinsights-${var.env}"
```
Comment (tomaszpowroznik) [inline infrastructure/main.tf:77]: it is created in line 13, should be same expression or best variable

locals {
  appInsightName = "${var.product}-${var.component}-appinsights-${var.env}"
}

and then ${appInsightName}

File: infrastructure/main.tf [hunk 6]
```diff
@@ -54,6 +54,41 @@ module "fpl-scheduler-db" {
...
+  alert_desc                 = "Web pages took longer than 1 seconds to load"
+  app_insights_query         = "requests | where url !contains '/health' | where success == 'True' | where duration > 1000"
+  custom_email_subject       = "Alert: performance errors"
+  frequency_in_minutes       = 5
+  time_window_in_minutes     = 5
+  severity_level             = "2"
+  action_group_name          = "fpla-support"
+  trigger_threshold_operator = "GreaterThan"
+  trigger_threshold          = 5
+  resourcegroup_name         = "fpl-case-service-${var.env}"
```
Comment (tomaszpowroznik) [inline infrastructure/main.tf:89]: duplicated in 66

Comment (Jamesnelson117) [reply infrastructure/main.tf:89]: Should be scoped to a different module no?

File: infrastructure/main.tf [hunk 7]
```diff
@@ -54,6 +54,41 @@ module "fpl-scheduler-db" {
...
+
+module "fpla-performance-alert" {
+  source                     = "git@github.com:hmcts/cnp-module-metric-alert"
+  location                   = "${var.appinsights_location}"
+
+  app_insights_name          = "${var.product}-${var.component}-appinsights-${var.env}"
+
+  alert_name                 = "fpla-bad-requests"
+  alert_desc                 = "Web pages took longer than 1 seconds to load"
+  app_insights_query         = "requests | where url !contains '/health' | where success == 'True' | where duration > 1000"
```
Comment (tomaszpowroznik) [inline infrastructure/main.tf:81]: and could be used instead on |where

**Pull request merged**